src/main/java/net/minecraftforge/client/animation/EntityAnimation.java

```diff
@@ -0,0 +1,174 @@
+     * <li>PASSIVE - DEFAULT</li>
+     * <li>PASSIVE - LAST</li>
+     * <li>ACTIVE - FIRST</li>
+     * <li>ACTIVE - DEFAULT (Cancelled if an ACTIVE - FIRST is executed)</li>
+     * <li>ACTIVE - LAST (Cancelled if an ACTIVE - FIRST/DEFAULT is executed)</li>
+     * </ul>
+     * @param o an entity animation
+     * @return See {@link Comparable} documentation
+     */
+    @Override
+    public final int compareTo(@NotNull EntityAnimation<E> o)
+    {
+        if (this.getMode() == o.getMode())
+        {
+            return this.getPriority().ordinal() - o.getPriority().ordinal();
+        }
+        else if (o.getMode() == Mode.ACTIVE)
+        {
+            return -1;
+        }
+        return 1;
+    }
+
+    /**
+     * The animation mode. Read individual enums for an explanation on which one to use.
+     */
+    public enum Mode
+    {
+        /**
+         * A passive animation is considered a background animation like walking or sitting. It is
+         * executed first before active animations. It can override or be overridden by other
+         * passive animations.
+         */
+        PASSIVE,
+
+        /**
+         * An active animation is considered an important animation. It will override passive
+         * animations. The big difference is that once an active animation is executed, all remaining
+         * active animations will not be executed.
+         */
+        ACTIVE
+    }
+
+    /**
+     * The priority of the animation. Read individual enums for an explanation on which one to use.
+     */
+    public enum Priority
+    {
+        /**
+         * Marks the animation to execute first. If the animation mode is {@link Mode#PASSIVE}, it
+         * may be overridden by animations with the same or later priority. Otherwise if the mode is
+         * {@link Mode#ACTIVE} and the animation is executed, active animations with the same or
+         * later priority will not be executed.
+         */
+        FIRST,
+
+        /**
+         * The default priority of animations. A passive animation executed with this priority can
+         * override any passive animations executed in an earlier priority. It is not guaranteed that
+         * an active animation will be executed at this priority if one was executed in an early
+         * priority.
+         */
+        DEFAULT,
+
+        /**
+         * Marks the animation to run last. A passive animation executed with this priority can
+         * override any passive animations executed in an earlier priority. It is not guaranteed that
+         * an active animation will be executed at this priority if one was executed in an early
+         * priority.
+         */
+        LAST
+    }
+
+    /**
+     * A record containing data used when calculating animations with the addition of partial ticks
+     */
+    protected record Context(float animateTicks, float animateSpeed, float bobAnimateTicks, float headYaw, float headPitch, float partialTick)
```

Comment:
I don't see why this record should exist.
It seems like the data in this record can just get passed in `apply()`  as parameters to avoid initialization every frame, and getting the data by doing `context.animateTicks`, `context.animateSpeed`, and so on is pretty verbose.

Reply:
I still have a few more things to include into `Context`. I think in this scenario it's much better to put everything into `Context` than have a method with 20 parameters. It's all optional data that can be used for creating animations. I can avoid initializing it every frame and just have persistent instance that gets updated before calling `apply`.

Reply:
I did not expect many more parameters to get added to the method, and a long strand of hard-to-read parameters does not sound pretty.
Avoiding initialization every frame by using a routinely updated persistence instance sounds good 👍

Reply:
Just to be more clear, I was exaggerating to 20 parameters. I wasn't sure on exactly the amount I would be adding. It would still end up being a readability problem.

Reply:
On second thought, I think `Context` needs to be more versatile. As it is right now, all the values (except partialTick) are speicific to a living entity. Maybe a system similar to `ModelComponent` where you can retrieve data with a key.